struct Test {
    init { self } // ok
    method1(self) => self // ok
}

self // error
do { self } // error
fn func() => self // error

%output

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #d4f48717de1e83dd>:6:1
  │
6 │ self // error
  │ ^^^^

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #d4f48717de1e83dd>:7:6
  │
7 │ do { self } // error
  │      ^^^^

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #d4f48717de1e83dd>:8:14
  │
8 │ fn func() => self // error
  │              ^^^^